src/docs/asciidoc/distributed_sql.adoc

```diff
@@ -0,0 +1,211 @@
+optimizes it using Apache Calcite. The result is an executable plan that
+is cached and reused by the subsequent executions of the same statement.
+
+The plan contains a tree of query fragments. A query fragment is a tree of
+operators that could be executed on a single member independently. Child
+fragments supply data to parent fragments, possibly through a network, until
+the root fragment is reached. The root fragment returns query results to a
+user.
+
+Only `IMap` objects are supported at the moment.
+
+=== Querying IMap
+
+==== Names
+
+The engine exposes `IMap` objects as tables in the predefined `partitioned`
+schema using exact names. This schema is in the engine search path so that
+`IMap` objects could be accessed with or without the schema name.
+
+Schema and table names are case sensitive. For example, the `IMap` with the name
+`employee` can be accessed as `employee` or `partitioned.employee`, but not as
+`Employee`.
+
+==== Fields
+
+The engine reads the first local entry of the `IMap` to get the list of fields
+accessible from the SQL. The engine extracts field metadata (name, type)
+from the key and the value, and combines them into a final list of fields.
+
+Field metadata extraction logic depends on the serialization format, as described
+below.
+
+-  For Java objects (`Serializable`, `DataSerializable`, `IdentifiedDataSerializable`),
+the object is deserialized if needed and then analyzed using reflection. Only public
+fields and getters are taken in count. A public field is exposed using its exact name.
+A getter is exposed using the JavaBean naming conventions. E.g. `getName` is exposed as
+`name` and `isActive` is exposed as `active`
+- For `Portable` objects, only the fields that are written in the `Portable.writePortable`
+method are exposed using their exact names
+- `HazelcastJsonValue` objects cannot be queried at the moment. This limitation will be
+removed in future releases
+
+Field names are case-sensitive.
+
+If the `IMap` does not have local entries on the member where the query is started,
+then the list of fields cannot be resolved, and an exception is thrown.
+
+We refer to the key and the value as *top-level* objects. The key is exposed as
+`__key` field, and the value is exposed as `this` field.
+
+If a top-level object does not have nested fields, then it is exposed as a normal
+field. Otherwise, it is exposed a *hidden field*. Hidden fields behave similarly
+to normal fields, but are not returned by `SELECT * ...` queries.
+
+If both the key and the value have fields with the same name, then the field of the
+value will be used.
+
+==== Indexes
+
+The engine could use `IMap` indexes to speed up the execution of certain queries.
+
+`SORTED` and `HASH` indexes are supported.
+
+==== HD
+
+`IMap` objects with the `InMemoryFormat.NATIVE` could only be queried if they have
+at least one index, and the property `hazelcast.hd.global.index.enabled` is set to
```

Comment:
Is this property set by default? If not, please mention it.

Reply:
Also include the <properties> code snippet how to do it in the configuraiton.
This is a temporary state, right? It won't be the case in future releases. Please mention this as well as I see someone being scared by that.

Reply:
The property is enabled by default. This is not a temporary state - we may have this restriction for many releases, if not forever. It could only be removed if we rewrite our HD storage to concurrent data structures.

The following changes are made:
1. Mentioned that `true` is the default
1. Added a link to the config section
1. Added the new property to the list of system properties